eng/testing/tests.mobile.targets

```diff
@@ -139,14 +139,17 @@
              TaskName="AppleAppBuilderTask"
              AssemblyFile="$(AppleAppBuilderTasksAssemblyPath)" />
 
+  <UsingTask Condition="'$(RunAOTCompilation)' == 'true' and ('$(TargetOS)' == 'MacCatalyst' or '$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'iOSSimulator' or '$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'tvOSSimulator')"
```

Comment:
Should we exclude Simulator if we are targeting devices? Do we have a device specifically for MacCatalyst

Reply:
No, I think stripping can apply to all apple configurations.  It'll save us a little bit of disk space.

Reply:
And there isn't a device for catalyst.  They run on macos directly.